src/lib/overlay-language-service-host.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import ts from 'typescript';
+import * as fs from 'fs';
+
+/**
+ * A LanguageServiceHost that overlays a versioned in-memory file collection
+ * on top of the filesystem. This host also caches files read from the
```

Comment:
I see `fileCache` working, but what is `files`? I don't see its contents being set.

Reply:
`files` is set from the outside. See `convertModule`